server/index.js

```diff
@@ -14,104 +15,68 @@ const loggingMiddleware = require('./logging');
-    props.initialData.transaction.tx_status
-  ) {
-    if (props.initialData.transaction.tx_status !== 'pending') {
-      return true;
-    }
-  } else {
-    return false;
-  }
-};
-
-const renderAndCache = app =>
-  async function (req, res, pagePath, queryParams) {
-    const { host } = req.headers;
-    // Define the cache key as you wish here:
-    const key = host + req.url;
-
-    // if page is in cache, server from cache
-    if (ssrCache.has(key)) {
-      console.info('SSR Response from cache for ', key);
-      res.setHeader('x-cache', 'HIT');
-      res.end(ssrCache.get(key));
-      return;
-    }
-
-    try {
-      /**
-       * Override res.end method before sending it to app.renderToHTML
-       * to be able to get the payload (renderedHTML) and save it to cache.
-       */
-      const _resEnd = res.end.bind(res);
-      res.end = function (payload) {
-        // Add here custom logic for when you do not want to cache the page, for example when
-        // the status is not 200
-        const canCache = getCanCache(pagePath, payload);
-
-        if (res.statusCode !== 200) {
-          console.warn('Oops, something is wrong, will skip the cache');
-        } else if (canCache) {
-          ssrCache.set(key, payload);
-        }
-        return _resEnd(payload);
-      };
-      // if not in cache, render the page into HTML
-      res.setHeader('x-cache', 'MISS');
-      console.info('SSR rendering without cache and try caching for ', key);
-      await app.renderToHTML(req, res, pagePath, queryParams);
-    } catch (err) {
-      app.renderError(err, req, res, pagePath, queryParams);
-    }
-  };
-
 app.prepare().then(() => {
   const server = express();
 
-  server.use(createPrometheusMiddleware({ app: server }));
-
   server.get(
     '/_next/',
     expressStaticGzip('/static/', {
       enableBrotli: true,
     })
   );
 
+  let routes = [];
+
+  // Store all the registered express routes for usage with metrics reporting
+  routes = expressListEndpoints(app).map(endpoint => ({
+    path: endpoint.path,
+    regexp: pathToRegexp(endpoint.path),
+  }));
+
+  // Manual route definitions for the proxied endpoints
+  routes.push({
+    path: '/txid/*',
+    regexp: /^\/txid(.*)/,
+  });
+  routes.push({
+    path: '/microblock/*',
+    regexp: /^\/microblock(.*)/,
+  });
+  routes.push({
+    path: '/block/*',
+    regexp: /^\/block(.*)/,
+  });
+  routes.push({
+    path: '/address/*',
+    regexp: /^\/address(.*)/,
+  });
+
   if (!dev) {
+    const promMiddleware = createPrometheusMiddleware({
+      options: {
+        normalizePath: path => {
+          // Get the url pathname without a query string or fragment
+          // (note base url doesn't matter, but required by URL constructor)
+          let pathTemplate = new URL(path, 'http://x').pathname;
+          // Match request url to the Express route
+          for (const pathRegex of routes) {
+            if (pathRegex.regexp.test(pathTemplate)) {
```

Comment:
This can throw, see try/catch here https://github.com/blockstack/stacks-blockchain-api/blob/ab45a151f4136a1e8907befc1d6047035f6fd6f8/src/api/init.ts#L92-L107